src/test/java/org/apache/accumulo/access/grammar/antlr/Antlr4Tests.java

```diff
@@ -0,0 +1,203 @@
+      public void syntaxError(Recognizer<?,?> recognizer, Object offendingSymbol, int line,
+          int charPositionInLine, String msg, RecognitionException e) {
+        super.syntaxError(recognizer, offendingSymbol, line, charPositionInLine, msg, e);
+        errors.incrementAndGet();
+      }
+
+    });
+    try {
+      Access_expressionContext ctx = parser.access_expression();
+      assertNull(ctx.exception);
+    } catch (RuntimeException e1) {
+      throw new AssertionError(e1);
+    }
+    assertEquals(0, errors.get());
+    assertEquals(0, parser.getNumberOfSyntaxErrors());
+  }
+
+  @Test
+  public void testCompareWithAccessExpressionImplParsing() throws Exception {
+    // This test checks that the parsing of the AccessExpressions in testdata.json
+    // using ANTLR have the same outcome as AccessExpression.of()
+    List<TestDataSet> testData = AccessEvaluatorTest.readTestData();
+    for (TestDataSet testSet : testData) {
+      for (TestExpressions test : testSet.tests) {
+        ExpectedResult result = test.expectedResult;
+        for (String cv : test.expressions) {
+          // System.out.println("Testing: " + cv);
+          if (result == ExpectedResult.ERROR) {
+            assertThrows(IllegalAccessExpressionException.class, () -> AccessExpression.of(cv));
+            assertThrows(AssertionError.class, () -> testParse(cv));
+          } else {
+            AccessExpression.of(cv);
+            testParse(cv);
+          }
+        }
+      }
+    }
+  }
+
+  @Test
+  public void testSimpleEvaluation() throws Exception {
+    String accessExpression = "(one&two)|(foo&bar)";
+    Authorizations auths = Authorizations.of("four", "three", "one", "two");
+    AccessExpressionAntlrEvaluator eval = new AccessExpressionAntlrEvaluator(List.of(auths));
+    assertTrue(eval.canAccess(accessExpression));
+  }
+
+  @Test
+  public void testSimpleEvaluationFailure() throws Exception {
+    String accessExpression = "(A&B&C)";
+    Authorizations auths = Authorizations.of("A", "C");
+    AccessExpressionAntlrEvaluator eval = new AccessExpressionAntlrEvaluator(List.of(auths));
+    assertFalse(eval.canAccess(accessExpression));
+  }
+
+  @Test
+  public void testCompareAntlrEvaluationAgainstAccessEvaluatorImpl() throws Exception {
+    // This test checks that the evaluation of the AccessExpressions in testdata.json
+    // using ANTLR have the same outcome as AccessEvaluatorImpl
+    List<TestDataSet> testData = AccessEvaluatorTest.readTestData();
+    for (TestDataSet testSet : testData) {
+      for (TestExpressions test : testSet.tests) {
+        for (String expression : test.expressions) {
+          List<Authorizations> authSets =
+              Stream.of(testSet.auths).map(Authorizations::of).collect(Collectors.toList());
+          AccessEvaluator evaluator = AccessEvaluator.builder().authorizations(authSets).build();
+
+          AccessExpressionAntlrEvaluator antlr = new AccessExpressionAntlrEvaluator(authSets);
```

Comment:
These can be moved outside of the loop.

```suggestion
        List<Authorizations> authSets =
              Stream.of(testSet.auths).map(Authorizations::of).collect(Collectors.toList());
        AccessEvaluator evaluator = AccessEvaluator.builder().authorizations(authSets).build();

        AccessExpressionAntlrEvaluator antlr = new AccessExpressionAntlrEvaluator(authSets);
        
        for (String expression : test.expressions) {
```

Reply:
Implemented change in 6fd8db5